projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-wpcom-online-subscription-service.php

```diff
@@ -113,6 +113,7 @@ public function is_current_user_pending_subscriber(): bool {
 	 * @return bool
 	 */
 	protected function user_can_view_content( $valid_plan_ids, $access_level, $is_blog_subscriber, $post_id ) {
+		$user_id = is_user_logged_in() ? wp_get_current_user()->ID : $this->user_id;
```

Comment:
I believe those are the same...

Reply:
The `$user_id` (which becomes `$this->user_id`) is only added this specific flow (where we know the user id but the user is not logged in), so we otherwise want to fallback on the using the existing `wp_get_current_user()->ID`. Also, if for some reason a user_id is provided but the user is logged in, we'd want to prefer the logged in user's ID. Under most other flows, the user would be logged in but `$this->user_id` would be null.